pkg/build/api/validation/validation.go

```diff
@@ -480,6 +471,15 @@ func validateTrigger(trigger *buildapi.BuildTriggerPolicy, fldPath *field.Path)
 			break
 		}
 		if trigger.ImageChange.From == nil {
+			if buildFrom == nil || buildFrom.Kind != "ImageStreamTag" {
+				invalidKindErr := field.Invalid(
+					fldPath.Child("imageChange"),
+					fmt.Sprintf("build from: %v", buildFrom),
+					"a default ImageChange trigger can only be used with builds based on ImageStreamTags.")
```

Comment:
"builds based on ImageStreamTags" may be confusing. I can't come up with better wording at the moment, but something to the effect of: "a build with a strategy that references an image that is not of type ImageStreamTag" (that definitely doesn't sound better :-), but would help you to know which reference to fix)